File: Java/Q40.java
import java.util.Scanner;

public class BinaryOctal {
    static String binaryToOctal(String binary) {
        int decimal = Integer.parseInt(binary, 2);
        return Integer.toOctalString(decimal);
    }

    static String octalToBinary(String octal) {
        int decimal = Integer.parseInt(octal, 8);
        return Integer.toBinaryString(decimal);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter binary number: ");
        String binary = sc.next();
        System.out.println("Octal: " + binaryToOctal(binary));

        System.out.print("Enter octal number: ");
        String octal = sc.next();
        System.out.println("Binary: " + octalToBinary(octal));
        sc.close();
    }
}

File: Python/Q40.py
# Q40.py - Program to Convert Binary to Octal and Vice Versa
num = input("Enter number: ")
base = int(input("Enter base (2 for binary, 8 for octal): "))
if base == 2:
    decimal = int(num, 2)
    print(f"Octal: {oct(decimal)[2:]}")
elif base == 8:
    decimal = int(num, 8)
    print(f"Binary: {bin(decimal)[2:]}")
else:
    print("Invalid base")